wallet-new/test/unit/Wallet/Inductive/Cardano.hs

```diff
@@ -159,18 +165,29 @@ equivalentT activeWallet esk = \mkWallet w ->
                 -> Utxo
                 -> TranslateT (EquivalenceViolation h) m HD.HdAccountId
     walletBootT ctxt utxo = do
-        res <- liftIO $ Kernel.createWalletHdRnd passiveWallet walletName
-                                                 spendingPassword assuranceLevel
-                                                 esk utxo
-
-        either createWalletErr (checkWalletAccountState ctxt) res
+        res <- liftIO $ Kernel.createWalletHdRnd passiveWallet
+                                                 False
+                                                 walletName
+                                                 assuranceLevel
+                                                 esk
+                                                 utxo
+        case res of
+             Left e -> createWalletErr (STB e)
+             Right hdRoot -> do
+                 let keystore = passiveWallet ^. Internal.walletKeystore
+                 liftIO $ Keystore.insert (WalletIdHdRnd $ hdRoot ^. HD.hdRootId) esk keystore
```

Comment:
Fixing this test highlighted something orthogonal but equally important: if, when applying a block, we discover that the wallet is in an inconsistent state (like in this case where we had a wallet in the acid-state but no corresponding key in the keystore) we should fail hard and not silently ignore the fact.

I have opened https://iohk.myjetbrains.com/youtrack/issue/CBR-341 to track this.